src/lazy/streaming_raw_reader.rs

```diff
@@ -319,6 +352,14 @@ impl<'a> IonInput for StdinLock<'a> {
     }
 }
 
+impl<T: Read, U: Read> IonInput for io::Chain<T, U> {
```

Comment:
🗺️ The `ion-cli` uses the `io::Chain` type to link together the first few bytes of an input stream (used to detect gzip, zstd, etc) with the rest of the stream.